src/components/OrgFinder/FindOrganization.tsx

```diff
@@ -74,53 +175,207 @@ class FindOrganization extends Component<Props, State> {
       .then((responseJSON) => {
         const { status } = responseJSON;
 
+        // if valid zipcode
         if (status === 'OK') {
-          const zipcodeLatLng = responseJSON.results[0].geometry.location;
+          const zipcodeLat = responseJSON.results[0].geometry.location.lat;
+          const zipcodeLng = responseJSON.results[0].geometry.location.lng;
+          const coordinateProps = {
+            lat: zipcodeLat,
+            lng: zipcodeLng,
+          }
+          const coordinate = new Coordinate(coordinateProps);
+          return coordinate;
+        } else {
           this.setState({
-            displayMap: true,
-            zipcodeLatLng,
+            displayMap: false,
+            displayError: true,
+            displayIcon: false,
           });
-        } else {
-          const {
-            alert,
-          } = this.props;
-          alert.show('Invalid zip code');
         }
       });
+    return;
   }
 
+  // haversine formula
+  getDistanceInKM(coordinate1, coordinate2): number {
```

Comment:
you can type annotate coordinate1 and coordinate2 like:
coordinate1: Coordinate, coordinate2: Coordinate